flink-staging/flink-hcatalog/src/main/scala/org/apache/flink/hcatalog/scala/HCatOutputFormat.scala

```diff
@@ -0,0 +1,449 @@
+        list.add(toJava(i17))
+        list.add(toJava(i18))
+        list.add(toJava(i19))
+      };
+      case Tuple20(i1: AnyRef, i2: AnyRef, i3: AnyRef, i4: AnyRef, i5: AnyRef, i6: AnyRef,
+      i7: AnyRef, i8: AnyRef, i9: AnyRef, i10: AnyRef, i11: AnyRef, i12: AnyRef, i13: AnyRef,
+      i14: AnyRef, i15: AnyRef, i16: AnyRef, i17: AnyRef, i18: AnyRef, i19: AnyRef, i20: AnyRef)
+      => {
+        list.add(toJava(i1))
+        list.add(toJava(i2))
+        list.add(toJava(i3))
+        list.add(toJava(i4))
+        list.add(toJava(i5))
+        list.add(toJava(i6))
+        list.add(toJava(i7))
+        list.add(toJava(i8))
+        list.add(toJava(i9))
+        list.add(toJava(i11))
+        list.add(toJava(i12))
+        list.add(toJava(i13))
+        list.add(toJava(i14))
+        list.add(toJava(i15))
+        list.add(toJava(i16))
+        list.add(toJava(i17))
+        list.add(toJava(i18))
+        list.add(toJava(i19))
+        list.add(toJava(i20))
+      };
+      case Tuple21(i1: AnyRef, i2: AnyRef, i3: AnyRef, i4: AnyRef, i5: AnyRef, i6: AnyRef,
+      i7: AnyRef, i8: AnyRef, i9: AnyRef, i10: AnyRef, i11: AnyRef, i12: AnyRef, i13: AnyRef,
+      i14: AnyRef, i15: AnyRef, i16: AnyRef, i17: AnyRef, i18: AnyRef, i19: AnyRef, i20: AnyRef,
+      i21: AnyRef) => {
+        list.add(toJava(i1))
+        list.add(toJava(i2))
+        list.add(toJava(i3))
+        list.add(toJava(i4))
+        list.add(toJava(i5))
+        list.add(toJava(i6))
+        list.add(toJava(i7))
+        list.add(toJava(i8))
+        list.add(toJava(i9))
+        list.add(toJava(i11))
+        list.add(toJava(i12))
+        list.add(toJava(i13))
+        list.add(toJava(i14))
+        list.add(toJava(i15))
+        list.add(toJava(i16))
+        list.add(toJava(i17))
+        list.add(toJava(i18))
+        list.add(toJava(i19))
+        list.add(toJava(i20))
+        list.add(toJava(i21))
+      };
+      case Tuple22(i1: AnyRef, i2: AnyRef, i3: AnyRef, i4: AnyRef, i5: AnyRef, i6: AnyRef,
+      i7: AnyRef, i8: AnyRef, i9: AnyRef, i10: AnyRef, i11: AnyRef, i12: AnyRef, i13: AnyRef,
+      i14: AnyRef, i15: AnyRef, i16: AnyRef, i17: AnyRef, i18: AnyRef, i19: AnyRef, i20: AnyRef,
+      i21: AnyRef, i22: AnyRef) => {
+        list.add(toJava(i1))
+        list.add(toJava(i2))
+        list.add(toJava(i3))
+        list.add(toJava(i4))
+        list.add(toJava(i5))
+        list.add(toJava(i6))
+        list.add(toJava(i7))
+        list.add(toJava(i8))
+        list.add(toJava(i9))
+        list.add(toJava(i11))
+        list.add(toJava(i12))
+        list.add(toJava(i13))
+        list.add(toJava(i14))
+        list.add(toJava(i15))
+        list.add(toJava(i16))
+        list.add(toJava(i17))
+        list.add(toJava(i18))
+        list.add(toJava(i19))
+        list.add(toJava(i20))
+        list.add(toJava(i21))
+        list.add(toJava(i22))
+      };
+      case _ =>
+        throw new IOException("Only scala tuple is allowed")
+
+    }
```

Comment:
We can reimplement this with converting scala collection to java collection.

```scala
o match {
      case Tuple1(i1: AnyRef) => Seq(toJava(i1)).asJava
      case Tuple2(i1: AnyRef, i2: AnyRef) => Seq(toJava(i1), toJava(i2)).asJava
      case Tuple3(i1: AnyRef, i2: AnyRef, i3: AnyRef) =>
        Seq(toJava(i1), toJava(i2), toJava(i3)).asJava
      case Tuple4(i1: AnyRef, i2: AnyRef, i3: AnyRef, i4: AnyRef) =>
        Seq(toJava(i1), toJava(i2), toJava(i3), toJava(i4)).asJava
      ...
}
```

Returning this value and passing this collection to `DefaultHCatRecord` constructor. With this implementation, we don't need to case `record` to `AnyRef`.